### ISSUE TITLE
How to implement cross entropy loss and class collisions removal?

### ISSUE THREAD

**Issue by linzhiqiu** [locuslab/FLYP#1]
Hey,

I am curious about how the cross entropy loss ablation study is implemented -- are you initializing the linear head with text features, or do you always use the current language encoder (per iteration) to extract text features as linear head?

Also, for class collision removal, do you remove the collided terms from the denominator of equation 1 in paper? 

Thanks!

**Comment by linzhiqiu** [comment]
One last question -- how do you perform weight ensembling? Do you used a fixed ratio 0.5 as in WiSE-FT? Or do you select the ratio based on the in-domain validation set performance?

**Comment by SachinG007** [comment]
Hi @linzhiqiu,

Sorry for the late reply.
1) For the cross-entropy ablation, it's the latter, we use the current language encoder to extract the text-features (for all the classnames) and use it as a linear head to get the logits from image representations.
2) For class collision removal, yes, we remove the terms from the denominator.
3) We performed weight ensembling with weights from [0.1, 0.2, .... 0.9, 1.0] and report the one corresponding to the best ID Val performance.

Let me know (better email) if something is not clear and I will be happy to help more.

**Comment by anhquancao** [comment]
Hi, could you point out to the code that you handle class collision ?
Thanks in advance.
